pkg/istiovalues/profiles.go

```diff
@@ -27,13 +28,22 @@ import (
 	"istio.io/istio/pkg/util/sets"
 )
 
-func ApplyProfiles(resourceDir string, version string, defaultProfile, userProfile string, userValues helm.Values) (helm.Values, error) {
+func ApplyProfilesAndPlatform(
+	resourceDir string, version string, platform config.Platform, defaultProfile, userProfile string, userValues helm.Values,
+) (helm.Values, error) {
 	profile := resolve(defaultProfile, userProfile)
 	defaultValues, err := getValuesFromProfiles(path.Join(resourceDir, version, "profiles"), profile)
 	if err != nil {
 		return nil, fmt.Errorf("failed to get values from profile %q: %w", profile, err)
 	}
-	return mergeOverwrite(defaultValues, userValues), nil
+	values := helm.Values(mergeOverwrite(defaultValues, userValues))
+
+	if platform != config.PlatformKubernetes && platform != config.PlatformUndefined {
```

Comment:
Apart from "openshift" (and kubernetes), are there any valid platforms used in Istio?

Reply:
Yes, there are also:
- k3d
- k3s
- microk8s
- minikube

See `profile-platform-*.yaml` files [here](https://github.com/istio/istio/tree/master/manifests/charts/istio-control/istio-discovery/files).